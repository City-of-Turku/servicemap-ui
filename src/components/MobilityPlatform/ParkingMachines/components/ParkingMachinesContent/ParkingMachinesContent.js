import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../styled/styled';
import TextComponent from '../../../TextComponent';

const ParkingMachinesContent = ({ item }) => {
  const intl = useIntl();

  /** For values that are not objects and do not contain localized strings */
  const singleValText = (messageId, value) => (
    <StyledTextContainer>
      <Typography component="p" variant="body2">
        {intl.formatMessage({ id: messageId }, { value })}
      </Typography>
    </StyledTextContainer>
  );

  const machineAddress = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  const formatPrice = price => price.toString().replace('.', ',');

  const parkingMachineInfo = (
    <StyledContainer>
      <StyledHeaderContainer>
        <Typography variant="subtitle1" component="h3">
          {intl.formatMessage({ id: 'mobilityPlatform.content.parkingMachine.title' })}
        </Typography>
      </StyledHeaderContainer>
      <StyledTextContainer>
        {item.address_fi ? <TextComponent messageId="mobilityPlatform.content.address" textObj={machineAddress} /> : null}
        <TextComponent messageId="mobilityPlatform.content.parkingMachine.location" textObj={item.extra.Sijainti} />
        {singleValText('mobilityPlatform.content.parkingMachine.payment', formatPrice(item.extra['Taksa/h']))}
        <TextComponent messageId="mobilityPlatform.content.parkingMachine.paymentTypes" textObj={item.extra.Maksutapa} />
        {item.extra.Muuta ? singleValText('mobilityPlatform.content.parkingMachine.otherInfo', item.extra.Muuta) : null}
      </StyledTextContainer>
    </StyledContainer>
  );

  return (
    <div>
      {parkingMachineInfo}
    </div>
  );
};

ParkingMachinesContent.propTypes = {
  item: PropTypes.shape({
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    extra: PropTypes.shape({
      Sijainti: PropTypes.objectOf(PropTypes.string),
      Maksutapa: PropTypes.objectOf(PropTypes.string),
      'Taksa/h': PropTypes.string,
      Muuta: PropTypes.string,
    }),
  }),
};

ParkingMachinesContent.defaultProps = {
  item: {},
};

export default ParkingMachinesContent;
