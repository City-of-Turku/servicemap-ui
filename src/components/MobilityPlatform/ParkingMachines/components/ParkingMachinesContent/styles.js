export default theme => ({
  padding: {
    padding: theme.spacing(0.7),
  },
  headerContainer: {
    width: '85%',
    borderBottom: '1px solid #000',
    paddingBottom: theme.spacing(0.5),
  },
  textContainer: {
    marginTop: theme.spacing(0.5),
  },
  margin: {
    margin: theme.spacing(0.4),
  },
});
