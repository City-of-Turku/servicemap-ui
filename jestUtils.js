import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });