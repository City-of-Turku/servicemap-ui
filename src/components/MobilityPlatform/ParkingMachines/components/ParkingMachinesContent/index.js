import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import ParkingMachinesContent from './ParkingMachinesContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(ParkingMachinesContent));
