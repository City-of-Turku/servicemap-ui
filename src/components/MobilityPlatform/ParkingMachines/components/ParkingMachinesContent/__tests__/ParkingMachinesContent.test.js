// Link.react.test.js
import React from 'react';
import ParkingMachinesContent from '../index';
import { initialState } from '../../../../../../redux/reducers/user';
import finnishTranslations from '../../../../../../i18n/fi';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  item: {
    address_fi: 'Testikatu',
    extra: {
      Sijainti: 'Katuosa',
      'Taksa/h': '1.8',
      Maksutapa: 'Kortti, k√§teinen',
      Muuta: 'Testiteksti',
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<ParkingMachinesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<ParkingMachinesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkingMachinesContent {...mockProps} />);

    const h6 = container.querySelectorAll('h6');
    const p = container.querySelectorAll('p');
    expect(h6[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.parkingMachine.title']);
    expect(p[0].textContent).toContain(`Osoite: ${mockProps.item.address_fi}`);
    expect(p[1].textContent).toContain(`Sijainti: ${mockProps.item.extra.Sijainti}`);
    expect(p[2].textContent).toContain(`Maksu: ${mockProps.item.extra['Taksa/h']}`);
    expect(p[3].textContent).toContain(`Maksutavat: ${mockProps.item.extra.Maksutapa}`);
    expect(p[4].textContent).toContain(`Muuta: ${mockProps.item.extra.Muuta}`);
  });
});
