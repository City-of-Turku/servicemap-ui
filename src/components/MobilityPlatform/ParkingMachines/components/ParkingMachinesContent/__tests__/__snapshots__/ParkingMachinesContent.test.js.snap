// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkingMachinesContent /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(ParkingMachinesContent)-container-1"
  >
    <div
      class="injectIntl(ParkingMachinesContent)-container-1"
    >
      <div
        class="injectIntl(ParkingMachinesContent)-headerContainer-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Pysäköintiautomaatti
        </h6>
      </div>
      <div
        class="injectIntl(ParkingMachinesContent)-textContainer-3"
      >
        <div
          class="injectIntl(ParkingMachinesContent)-margin-5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Osoite: Testikatu
          </p>
        </div>
        <div
          class="injectIntl(ParkingMachinesContent)-margin-5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Sijainti: Katuosa
          </p>
        </div>
        <div
          class="injectIntl(ParkingMachinesContent)-margin-5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Maksu: 1.8 €/t
          </p>
        </div>
        <div
          class="injectIntl(ParkingMachinesContent)-margin-5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Maksutavat: Kortti, käteinen
          </p>
        </div>
        <div
          class="injectIntl(ParkingMachinesContent)-margin-5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Muuta: Testiteksti
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
